### PR TITLE
[FINAL] [BOOM!] - Retina Text

### PR DESCRIPTION
**Pull request by aab29**
:boom: 

Retina text works.

Cherry-picked Isaac's commits from https://github.com/IkeBart/moai-dev/tree/ike-font-op-size-exp .  I _think_ I got them all right.

The new code fixes issues with string bounds calculation when a glyphScale is applied to them.  It also fixes an issue where the font size returned for single-line text was sometimes too large.

:boom:

### CODE REVIEW

File: src/moaicore/MOAIFont.cpp [hunk 1]
```diff
@@ -568,7 +567,7 @@ float MOAIFont::OptimalSize ( cc8* text, float width, float height, float minSiz
 	
 	
 	MOAITextBox *textBox = new MOAITextBox ();
-	textBox->SetRect ( 0.0f, 0.0f, textLength * maxSize * 2, maxSize * 2 );
+	textBox -> SetRect(0.0f, 0.0f, textLength * maxSize * 20, maxSize * 20);
```
Comment (jeiting) [inline src/moaicore/MOAIFont.cpp]: Numbers, they be magic.

Comment (aab29) [inline src/moaicore/MOAIFont.cpp]: :tophat: :rabbit2:

Not sure where that comes from or why it works.  I'll ask Isaac to explain it when I get a chance.

**Comment by NachoSoto** [comment]
Example of how important this is: https://mindsnacks.atlassian.net/browse/GAMES-274

File: src/moaicore/MOAIFont.cpp (hunk 1 above)
Comment (alexey-dc) [inline src/moaicore/MOAIFont.cpp]: Looks like he's using the same trick I used on the Lua side - first make sure the text fits in a render box, then get the exact size by using BoundsForRange (that's what he proceeds to use further down in this function). So 20 is really a number that is supposed to guarantee a box big enough to fit the text, and later down that size is discarded.

Comment (NachoSoto) [inline src/moaicore/MOAIFont.cpp]: Why 20 and not 17.5? #MagicNumbers

**Comment by alexey-dc** [comment]
:shipit: 
Adresses our immediate use case, we can clean it up later if we're inclined.

**Pull request merged**